pkg/framework/webhooks.go

```diff
@@ -0,0 +1,160 @@
+	key := client.ObjectKey{Name: name}
+	existing := &admissionregistrationv1.MutatingWebhookConfiguration{}
+
+	if err := wait.PollImmediate(RetryShort, WaitShort, func() (bool, error) {
+		if err := c.Get(context.TODO(), key, existing); err != nil {
+			klog.Errorf("Error querying api for MutatingWebhookConfiguration object %q: %v, retrying...", name, err)
+			return false, nil
+		}
+		return true, nil
+	}); err != nil {
+		return nil, fmt.Errorf("error getting MutatingWebhookConfiguration %q: %v", name, err)
+	}
+	return existing, nil
+}
+
+// GetValidatingWebhookConfiguration gets ValidatingWebhookConfiguration object by name
+func GetValidatingWebhookConfiguration(c client.Client, name string) (*admissionregistrationv1.ValidatingWebhookConfiguration, error) {
+	key := client.ObjectKey{Name: name}
+	existing := &admissionregistrationv1.ValidatingWebhookConfiguration{}
+
+	if err := wait.PollImmediate(RetryShort, WaitShort, func() (bool, error) {
+		if err := c.Get(context.TODO(), key, existing); err != nil {
+			klog.Errorf("Error querying api for ValidatingWebhookConfiguration object %q: %v, retrying...", name, err)
+			return false, nil
+		}
+		return true, nil
+	}); err != nil {
+		return nil, fmt.Errorf("error getting ValidatingWebhookConfiguration %q: %v", name, err)
+	}
+	return existing, nil
+}
+
+// DeleteValidatingWebhookConfiguration deletes the specified ValidatingWebhookConfiguration object
+func DeleteValidatingWebhookConfiguration(c client.Client, webhookConfiguraiton *admissionregistrationv1.ValidatingWebhookConfiguration) error {
+	return wait.PollImmediate(RetryShort, WaitShort, func() (bool, error) {
+		if err := c.Delete(context.TODO(), webhookConfiguraiton); apierrors.IsNotFound(err) {
+			return true, nil
+		} else if err != nil {
+			klog.Errorf("error querying api for ValidatingWebhookConfiguration object %q: %v, retrying...", webhookConfiguraiton.Name, err)
+			return false, nil
+		}
+		return true, nil
+	})
+}
+
+// DeleteMutatingWebhookConfiguration deletes the specified MutatingWebhookConfiguration object
+func DeleteMutatingWebhookConfiguration(c client.Client, webhookConfiguraiton *admissionregistrationv1.MutatingWebhookConfiguration) error {
+	return wait.PollImmediate(RetryShort, WaitShort, func() (bool, error) {
+		if err := c.Delete(context.TODO(), webhookConfiguraiton); apierrors.IsNotFound(err) {
+			return true, nil
+		} else if err != nil {
+			klog.Errorf("error querying api for MutatingWebhookConfiguration object %q: %v, retrying...", webhookConfiguraiton.Name, err)
+			return false, nil
+		}
+		return true, nil
+	})
+}
+
+// UpdateMutatingWebhookConfiguration updates the specified mutating webhook configuration
+func UpdateMutatingWebhookConfiguration(c client.Client, updated *admissionregistrationv1.MutatingWebhookConfiguration) error {
+	return wait.PollImmediate(RetryShort, WaitMedium, func() (bool, error) {
```

Comment:
why `WaitMedium` here?
this is `GetMutatingWebhookConfiguration` and so nesting `wait.PollImmediate` calls.

I think the only purpose of using `wait.PollImmediate` in this library funcs is to protect and don't fail the test on a transient network issue? to that end we could may be just wait a few seconds? wdyt?

Reply:
Sounds good to me. Webhooks are ready to react as soon as they are updated in the cluster. I could see potential to use `apply` method from library-ho, but short wait should be enough in any case.

Reply:
@enxebre I decreased both wait periods to be at `1 min` at max, in case of the client throttling appears, and `3 min` for waiting on webhook configuration to be synced, PTAL